tests/unit/test_dataset.py

```diff
@@ -136,21 +137,21 @@ def test_const_dataset__uniform_batch_data(dataset_type, batch_size):
 
 def test_const_dataset__sparse_batch_data(dataset_type):
     batch_size = 3
-    components = {"node": 3, "sym_load": 2, "asym_load": 4, "link": 4}
+    components = {ComponentType.node: 3, ComponentType.sym_load: 2, ComponentType.asym_load: 4, ComponentType.link: 4}
     data = {
-        "node": {
+        ComponentType.node: {
             "data": np.zeros(shape=3, dtype=power_grid_meta_data[dataset_type]["node"]),
             "indptr": np.array([0, 2, 3, 3]),
         },
-        "sym_load": {
+        ComponentType.sym_load: {
             "data": np.zeros(shape=2, dtype=power_grid_meta_data[dataset_type]["sym_load"]),
             "indptr": np.array([0, 0, 1, 2]),
         },
-        "asym_load": {
+        ComponentType.asym_load: {
             "data": np.zeros(shape=4, dtype=power_grid_meta_data[dataset_type]["asym_load"]),
             "indptr": np.array([0, 2, 3, 4]),
         },
-        "link": np.zeros(shape=(batch_size, 4), dtype=power_grid_meta_data[dataset_type]["link"]),
+        ComponentType.link: np.zeros(shape=(batch_size, 4), dtype=power_grid_meta_data[dataset_type]["link"]),
     }
```

Comment:
This relates to one of my previous comment: this mixed way of doing at component level but not on attribute level could be confusing.

Reply:
@Jerry-Jinfeng-Guo I understand your point of view. I guess, as pointed out by @mgovers, we should make some design decisions if we want to also include some sort of `AttributeType`. Since it is out of scope for now, I will resolve and open a new ticket in the backlog.